### DISCUSSION TITLE
Help with relationship

### DISCUSSION THREAD

**Discussion by SupertigerDev** [prisma/prisma#14812] [Q&A]
```prisma
model User {
  id                  Int     @id @default(autoincrement())
  username            String

  servers             Server[] // idk what to do??
  
  memberInServers     ServerMember[]
  serversCreatedByMe  Server[] @relation("server_creator")
}

model Server {
  id             Int      @id @default(autoincrement())
  name           String

  serverMembers  ServerMember[]
  createdById    Int 
  createdBy      User @relation(fields: [createdById], references: [id], name: "server_creator")

}

model ServerMember {
  id       Int      @id @default(autoincrement())
  serverId Int
  userId   Int

  server   Server @relation(fields: [serverId], references: [id])
  user     User @relation(fields: [userId], references: [id])
}
```
How can i make `Server[]` be available in `User` model? I want a list of all the servers that the user has joined. This is indicated if they are in the `ServerMember` model or not. I could just use `memberInServers` but I was wondering if there was a way to access a list of `Server[]` rather than a list of `ServerMember[]`

Note: I am migrating from mongoose to prisma and postgresql, I don't really wanna change the structure of my database too much.

**Comment by SupertigerDev** [comment]
Any help?

**Comment by nurul3101** [comment]
Hey @Supertigerr 👋 

Your schema represents [explicit many to many](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#explicit-many-to-many-relations) relations with a connecting table, `ServerMember` in this case. 
If you don't want to use this connecting table, you can use [implicit many to many](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations) relations. 
Here's an example of using implicit relations

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model User {
  id                 Int      @id @default(autoincrement())
  username           String
  serversJoinedByMe  Server[]
  serversCreatedByMe Server[] @relation("server_creator")
}

model Server {
  id          Int    @id @default(autoincrement())
  name        String
  createdById Int
  createdBy   User   @relation(fields: [createdById], references: [id], name: "server_creator")
  joinedUsers User[]
}
```

**Reply by SupertigerDev** [reply]
Thanks for the reply. But What if I also need `ServerMember`? Because I will add additional data such as joined date, permission number, etc

**Reply by pantharshit00** [reply]
Hello @Supertigerr 

Unfortunately that won't be possible. You can maybe open a feature request where we can figure out how we can try to skip the relation table in the API if the user want it. You can open it here: https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
